### DISCUSSION TITLE
YouTube is testing blocking access?

### DISCUSSION THREAD

**Discussion by kaneryu** [sigma67/ytmusicapi#599] [General]
https://x.com/justusecobalt/status/1799043789967036620?t=eAGsZEoz-aahh8ZZ9mTdoA&s=19

Has anyone noticed anything failing a lot recently? I haven't (yet).

Does this affect this lib at all?

**Comment by theyak** [comment]
I've done about 5,000 requests to `get_song()` over a period of 8 months. I got blocked today. Oddly, other things, like `get_artist()` are still working. Only a handful of requests in the past few days, but today I got blocked.

Here's what the output looks like.

```
{'playabilityStatus': {'status': 'LOGIN_REQUIRED', 'reason': 'Sign in to confirm you’re not a bot', 'errorScreen': {'playerErrorMessageRenderer': {'subreason': {'runs': [{'text': 'This helps protect our community. '}, {'text': 'Learn more', 'navigationEndpoint': {'clickTrackingParams': 'CAAQu2kiEwjWu_aQ36-HAxUjkOUHHSKeAvs=', 'urlEndpoint': {'url': 'https://support.google.com/youtube/answer/3037019#zippy=%2Ccheck-that-youre-signed-into-youtube'}}}]}, 'reason': {'runs': [{'text': 'Sign in to confirm you’re not a bot'}]}, 'proceedButton': {'buttonRenderer': {'style': 'STYLE_PRIMARY', 'isDisabled': False, 'text': {'simpleText': 'Sign in'}, 'navigationEndpoint': {'clickTrackingParams': 'CAEQptEGIhMI1rv2kN-vhwMVI5DlBx0ingL7', 'signInEndpoint': {'nextEndpoint': {'clickTrackingParams': 'CAEQptEGIhMI1rv2kN-vhwMVI5DlBx0ingL7', 'urlEndpoint': {'url': ''}}}}, 'trackingParams': 'CAEQptEGIhMI1rv2kN-vhwMVI5DlBx0ingL7'}}, 'thumbnail': {'thumbnails': [{'url': '//s.ytimg.com/yts/img/meh7-vflGevej7.png', 'width': 140, 'height': 100}]}, 'icon': {'iconType': 'ERROR_OUTLINE'}}}, 'contextParams': 'Q0FFU0FnZ0I='}}
```

**Reply by sigma67** [reply]
Related: https://github.com/yt-dlp/yt-dlp/issues/10128
https://github.com/iv-org/invidious/issues/4734

Are you using oauth or browser auth?

**Reply by theyak** [reply]
This is only when not logged in. Once logged in with authentication, things work fine. But get_song(), in the past, hasn't required authentication.

**Comment by svioletg** [comment]
I've been getting this exact "Sign in to confirm you're not a bot" issue when trying to use yt-dlp at all. I've found some old issues referring to it but no definitive solutions.
